src/script/hooks/useHideElement.ts

```diff
@@ -62,23 +63,23 @@ const useHideElement = (timeout: number, skipClass?: string) => {
       }
     };
 
-    ref.current.addEventListener('mouseleave', hideElement);
-    ref.current.addEventListener('mouseout', hideElement);
-    ref.current.addEventListener('mousemove', onMouseMove);
-    ref.current.addEventListener('keydown', onKeyDown);
+    ref.addEventListener('mouseleave', hideElement);
+    ref.addEventListener('mouseout', hideElement);
+    ref.addEventListener('mousemove', onMouseMove);
+    ref.addEventListener('keydown', onKeyDown);
 
     startTimer();
     return () => {
       window.clearTimeout(hideTimeout);
-      ref.current.removeEventListener('mouseleave', hideElement);
-      ref.current.removeEventListener('mouseout', hideElement);
-      ref.current.removeEventListener('mousemove', onMouseMove);
-      ref.current.removeEventListener('keydown', onKeyDown);
-      ref.current.classList.remove(hideControlsClass);
+      ref.removeEventListener('mouseleave', hideElement);
+      ref.removeEventListener('mouseout', hideElement);
+      ref.removeEventListener('mousemove', onMouseMove);
+      ref.removeEventListener('keydown', onKeyDown);
+      ref.classList.remove(hideControlsClass);
     };
-  }, [ref.current]);
```

Comment:
Listening to a ref change was never supposed to work. 
Somehow react18 made this more obvious as the teardown was called later on and the ref was already `null` without being teardown